### PR TITLE
#165762957 User can add Books 

### PR DESCRIPTION
**Pull request by tolumide-ng**
#### What does this PR do?
User can add books

#### Description of Task to be completed?
- Adds test for possible scenario of user can add books
- Adds test for default route to this hellobooks
- Updates .env.sample file
- Edits clients on knexfile.js and adds directory for migrations on test environment
- Adds Controller for add books feature
- Adds migrations for the books table
- Adds endpoint for the user can add books feature
- Adds mockData file for test details

#### How should this be manually tested?
1. Clone the repo and cd into it
2. Use **npm install** to install dependencies
3. Add all necessary environment variables in your local .env file using the .env.sample file
4. Create the databases locally and run the migrations command for development and test
5. **npm run test** to run tests
6. **npm run dev** to run the project locally on your machine

#### Any background context you want to provide?
Provide all the needed environment variables and ensure run migrations

#### What are the relevant pivotal tracker stories?
#165762957

### CODE REVIEW

File: server/middlewares/controllers/bookControllers.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+// const Book = require('../../models/Book');
+const knexfile = require("../../../knexfile")[process.env.NODE_ENV];
+const knex = require("knex")(knexfile);
+
+class BookController {
+  static async addBooks(req, res) {
+    const { title, coverType, description, isbn, publisher, year } = req.body;
+
+    // await Book
```
Comment (bahdcoder) [inline server/middlewares/controllers/bookControllers.js]: Why do we have this comments ? This looks like the right way to go. Isn't it working ?

File: server/middlewares/controllers/bookControllers.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    //   .query()
+    //   .insert({ title, coverType, description, isbn, publisher, year });
```
Comment (bahdcoder) [inline server/middlewares/controllers/bookControllers.js]: Oh, and also, don't allow commented code in general.

File: server/middlewares/controllers/bookControllers.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+      coverType,
+      description,
+      isbn,
+      publisher,
+      year
+    });
+
+    return res
+      .status(201)
+      .jsend({ title, message: "Book has been added to the library" });
```
Comment (bahdcoder) [inline server/middlewares/controllers/bookControllers.js]: Let this respond with the newly added book.

File: server/middlewares/controllers/bookControllers.js (hunk 1 above)
Comment (tolumide-ng) [reply server/middlewares/controllers/bookControllers.js]: Yes, it doesn't work when I try to implement it that way. I would appreciate if you could help look through it

**Comment by bahdcoder** [comment]
Closing in favour of updated branch.

**Pull request closed without merging**